force-app/main/default/classes/SOQL.cls

```diff
@@ -270,92 +356,182 @@ public with sharing class SOQL {
+    List<String> emailDomainList = new List<String>(emailDomains);
+    
+    // Build the dynamic SOQL query
+    String query = 'SELECT Id, Name, Website FROM Account WHERE ';
+    
+    // Add conditions for each domain
+    List<String> conditions = new List<String>();
+    for (String domain : emailDomainList) {
+        conditions.add('Website LIKE \'%' + domain + '%\'');
+    }
+    
+    // Join the conditions with 'OR'
+    query += String.join(conditions, ' OR ');
+    system.debug('this is query => '+query);
+    // Execute the query
+    List<Account> matchingAccounts = Database.query(query);
+    
+    // Debug the matching accounts
+    system.debug('Matching Accounts: ' + matchingAccounts);
+    
+    return matchingAccounts;
 
-    /**
-     * Question 23
-     * Clones and Inserts Opportunity Line Items from an original Opportunity to a list of cloned Opportunities.
-     * Adds a randomized number between 1 to 3 to the quantity of each line item.
-     * Example: cloneLineItems('001XXXXXXXXXXXX', [list of opportunity ids]) should clone the line items from the Opportunity with the Id to each Opportunity in the list
-     * @param Original Opportunity Id
-     * @param List of cloned Opportunity Ids
-     */
-    public static void cloneLineItems(Id origOppId, List<Id> cloneOppIds) {
-        // Query the original Opportunity Line Items
-
-        // Create a list to hold the new line items
-
-        // Generate a random number between 1 and 3
-
-        // Loop through each cloned Opportunity Id
+}
 
-            // Loop through each original line item
 
-                // Clone the original line item and add the random number to the quantity
+     /**
+* Question 23
+* Clones and Inserts Opportunity Line Items from an original Opportunity to a list of cloned Opportunities.
+* Adds a randomized number between 1 to 3 to the quantity of each line item.
+* Example: cloneLineItems('001XXXXXXXXXXXX', [list of opportunity ids]) should clone the line items from the Opportunity with the Id to each Opportunity in the list
+* @param Original Opportunity Id
+* @param List of cloned Opportunity Ids
+*/
+public static void cloneLineItems(Id origOppId, List<Id> cloneOppIds) {
+    // Query for the line items related to the source opportunity
+    List<OpportunityLineItem> sourceLineItems = [
+        SELECT Quantity, PricebookEntryId, UnitPrice
+        FROM OpportunityLineItem
+        WHERE OpportunityId = :origOppId
+    ];
+
+    // Prepare a list to hold the cloned line items for bulk insertion
+    List<OpportunityLineItem> clonedLineItems = new List<OpportunityLineItem>();
+
+    // Loop through each target opportunity Id
+    for (Id targetOpportunityId : cloneOppIds) {
+        // For each source line item, create a clone for the target opportunity
+        for (OpportunityLineItem sourceItem : sourceLineItems) {
+            OpportunityLineItem clonedItem = sourceItem.clone(false, true); // shallow clone, preserve read-only fields
```

Comment:
Your comment on this line contradicts your code. It states it's a shallow clone, but it appears (based on the 'true' parameter) that it's a deep clone. Just be careful using the `clone()` method since changes you make to the clone (eg, OpportunitId field) can change the original record if not a deep clone.